### PR TITLE
Lint test files and ignore `@ts-ignore` rule within them.

### PR DESCRIPTION
**Pull request by threehams**
During the last PR, I noticed that lint errors showed up in-editor, but this wasn't reflected in CI. It took me a bit of time to figure out if this was going to cause the CI checks to fail.

This just overrides the lint rules to allow `ts-ignore` within tests (and allows for other rules to be suppressed over time, as well). That should prevent the disconnect between code and tests for new contributors.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -20,6 +20,15 @@ module.exports = {
     'react-hooks/exhaustive-deps': 'error',
     'no-console': 'error',
   },
+  overrides: [
+    {
+      files: ['*.test.ts', '*.test.tsx'],
+      rules: {
+        // Allow testing runtime errors when not using TS
```
Comment (JeromeDeLeon) [inline .eslintrc.js:27]:
```suggestion
        // Allow testing runtime errors to suppress TS errors
```

**Review by JeromeDeLeon — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (JeromeDeLeon) [reply .eslintrc.js:27]: I'll allow this, but we should limit using `@ts-ignore` as much as possible.

Comment (bluebill1049) [reply .eslintrc.js:27]: agree!

**Comment by threehams** [comment]
hm. I looked at all the `ts-ignore` usages, and none of them are being used for testing invalid input (like I thought they might). I'll get this ready, but then get rid of all the `ts-ignore` in the next commit.

**Comment by bluebill1049** [comment]
awesome thanks @threehams! @JeromeDeLeon you happy to approve this PR?

**Review by JeromeDeLeon — APPROVED** [review]

**Pull request merged**